Engineroom/engineroom.cpp

```diff
@@ -0,0 +1,135 @@
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <cstdlib>
+#include <ctime>
+
+using namespace std;
+
+// Function to read text from a file and append a newline
+string readFromFile(const string& filename)
+{
+    ifstream file(filename);
+    if (!file)
+    {
+        cerr << "Error: Could not open file " << filename << endl;
+        return "";
+    }
+    string content;
+    getline(file, content, '\0'); // Read entire file content into string
+    file.close();
+    content += "\n"; // Append a newline to the content
+    return content;
+}
+
+// Class for the Engine Room
+class EngineRoom
+{
+public:
+    // Function to handle entering the engine room
+    void enter()
+    {
+        cout << readFromFile("engineroom_entry.txt") << endl;
+
+        int choice;
+        do {
+            cout << "Choose an option:\n";
+            cout << "1. Look at the engine\n";
+            cout << "2. Leave the room\n";
+            cout << "Enter your choice: ";
+            cin >> choice;
+            cout << "\n"; // Add a blank line after input
+
+            switch (choice)
+            {
+            case 1:
+                lookAtEngine();
+                break;
+            case 2:
+                cout << readFromFile("engineroom_leave.txt") << endl;
+                return;
+            default:
+                cout << "Invalid choice. Please try again.\n\n"; // Double newline for clarity
+            }
+        } while (choice != 2);
+    }
+
+private:
+    bool enginePiece = false; // Assuming initially player doesn't have engine piece

```

Comment:
Class looks great! This variable is the one change I would request. The player class was planned to hold the engine piece variable, so that can be referenced in the enter function where needed. The enter function will likely need to take in the player object as an argument.